gm4_scuba_gear/data/gm4_scuba_gear/functions/check_gear.mcfunction

```diff
@@ -1,2 +1,5 @@
+# @s = player in water
+# run from main
+
 execute unless entity @s[predicate=gm4_scuba_gear:tank_equipped] run function gm4_scuba_gear:remove_effect
-execute at @s unless block ~ ~1 ~ #gm4:water unless block ~ ~1 ~ #gm4:waterloggable[waterlogged=true] run function gm4_scuba_gear:remove_effect
+execute at @s anchored eyes positioned ^ ^ ^ unless predicate gm4_scuba_gear:in_water run function gm4_scuba_gear:remove_effect
```

Comment:
What was the reasoning for using a predicate here instead of what I thought was a faster block check?

Reply:
This was discussed in the discord at https://discord.com/channels/151141188961828864/801191203966156891/872333289745158174
With no notable performance difference between predicate and block check, the predicate is preferable for future proofing